### ISSUE TITLE
Method filter() of QuerySet receives a plain object

### ISSUE THREAD

**Issue by murtukov** [tommikaikkonen/redux-orm#224]
The documentation says that the method **filter** of **QuerySet** class can take a callback function which in its turn takes the instance of the model as its sole argument, but in fact the function gets a plain object:

```
Post.all().filter(post => {
    post // is a plain object
})
```

So instead of the code above I have to write the following:
```
Post.all().toModelArray.filter(post => {
    post // is a plain object
})
```

And other question (maybe its not the right place to write about it, but i find it relevant to the topic). Somehow the method filter() could be used directly on models:
```
Post.filter(post => {
    post // is a plain object
})
```

**Comment by haveyaseen** [comment]
Yes, exactly right. Because `filter()` operates [at the database level](https://github.com/tommikaikkonen/redux-orm/blob/cd9c71490388bc2374a3b1e38b460e6c67bd1e6b/src/db/Table.js#L139) it receives a reference, not a Model instance. Would happily accept a PR :slightly_smiling_face:
